### PR TITLE
Add `ror` key for instances of entity 

### PR DESCRIPTION
**Pull request by jspaaks**
**Related issues**

Refs:

- #480

**Describe the changes made in this pull request**

In this PR:
- added a `ror` key to all entity instances (via `$defs.entity`)
- added a test (for `authors`) under 1.3.0
- added `ror` entries to the schema guide

Used regex from here: https://github.com/ror-community/ror-schema/blob/a985c6e97293c0ebcc50aae89a2092c557c1dbec/ror_schema_v2_0.json#L116

**Review checklist**

- [x] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [x] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [x] Please run tests locally.

```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout <branch>
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
<do other things>
```

- [ ] `<do other things>`

### CODE REVIEW

**Review by sdruskat — COMMENTED** [review]
Thanks for this work!

There is one thing I'd like to discuss: On the topic of the ROR ID, the [ROR docs](https://ror.readme.io/docs/ror-identifier-pattern) say

> The preferred form of a ROR identifier (a "ROR ID", the id element in a ROR record) is the entire URL: https://ror.org/02mhbdp94. However, the ROR API will also recognize forms such as ror.org/02mhbdp94 or 02mhbdp94, so these forms of the ROR ID may also be used.

This is similar to ORCID (which is basically "the same thing" for individuals), where the correct identifier is also the URL.

Hence I think I'd prefer CFF 1.3.0 to take the complete URL as pattern, not just the unique part. Also for the sake of consistency (with ORCID).

If you agree, @jspaaks, I can put in the necessary changes.

File: schema.json [hunk 1]
```diff
@@ -1843,6 +1847,18 @@
...
+        "ror": {
+            "$comment": "Regex from 'https://github.com/ror-community/ror-schema/blob/a985c6e97293c0ebcc50aae89a2092c557c1dbec/ror_schema_v2_0.json#L116'",
+            "description": "An identifier for a research organization. See https://ror.org/ for more information.",
+            "examples": [
+                "04bwf3e34",
+                "00rbjv475"
+            ],
+            "maxLength": 9,
+            "minLength": 9,
+            "pattern": "^0[0-9|a-z]{8}$",
```
Comment (jspaaks) [inline schema.json:1859]:
```suggestion
            "pattern": "^https://ror.org/0[0-9|a-z]{8}$",
```

File: schema-guide.md [hunk 2]
```diff
@@ -3998,6 +4014,20 @@ When there are multiple licenses, it is assumed their relationship is OR, not AN
...
+### `definitions.ror`
+
+- **type**: string of length 9 with pattern `^0[0-9|a-z]{8}$`
+- **required**: `false`
+- **description**: The entity's Research Organization Registry identifier, see https://ror.org.
+- **usage**:<br><br>
+    ```yaml
+    authors:
+    - name: German Aerospace Center
+      ror: 04bwf3e34
```
Comment (jspaaks) [inline schema-guide.md:4026]:
```suggestion
      ror: https://ror.org/04bwf3e34
```

File: examples/1.3.0/pass/entity-author-with-ror/CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,9 @@
+authors:
+- name: German Aerospace Center
+  ror: 04bwf3e34
+- name: Netherlands eScience Center
+  ror: 00rbjv475
```
Comment (jspaaks) [inline examples/1.3.0/pass/entity-author-with-ror/CITATION.cff:5]:
```suggestion
  ror: https://ror.org/04bwf3e34/00rbjv475
```

File: examples/1.3.0/pass/entity-author-with-ror/CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,9 @@
+authors:
+- name: German Aerospace Center
+  ror: 04bwf3e34
+- name: Netherlands eScience Center
+  ror: https://ror.org/04bwf3e34/00rbjv475
```
Comment (jspaaks) [inline examples/1.3.0/pass/entity-author-with-ror/CITATION.cff:5]:
```suggestion
  ror: https://ror.org/00rbjv475
```

File: schema-guide.md [hunk 5]
```diff
@@ -1177,6 +1179,20 @@ authors:
...
+### `definitions.entity.ror`
+
+- **type**: [`definitions.ror`](#definitionsror).
+- **required**: `false`
+- **description**: The entity's Research Organization Registry identifier, see https://ror.org.
+- **usage**:<br><br>
+    ```yaml
+    authors:
+    - name: German Aerospace Center
+      ror: 04bwf3e34
```
Comment (jspaaks) [inline schema-guide.md:1191]:
```suggestion
      ror: https://ror.org/04bwf3e34
```

File: schema-guide.md [hunk 6]
```diff
@@ -1177,6 +1179,20 @@ authors:
...
+- **type**: [`definitions.ror`](#definitionsror).
+- **required**: `false`
+- **description**: The entity's Research Organization Registry identifier, see https://ror.org.
+- **usage**:<br><br>
+    ```yaml
+    authors:
+    - name: German Aerospace Center
+      ror: https://ror.org/04bwf3e34
+    - name: Netherlands eScience Center
+      ror: 00rbjv475
```
Comment (jspaaks) [inline schema-guide.md:1193]:
```suggestion
      ror: https://ror.org/00rbjv475
```

File: schema-guide.md [hunk 7]
```diff
@@ -3998,6 +4014,20 @@ When there are multiple licenses, it is assumed their relationship is OR, not AN
...
+- **type**: string of length 9 with pattern `^0[0-9|a-z]{8}$`
+- **required**: `false`
+- **description**: The entity's Research Organization Registry identifier, see https://ror.org.
+- **usage**:<br><br>
+    ```yaml
+    authors:
+    - name: German Aerospace Center
+      ror: https://ror.org/04bwf3e34
+    - name: Netherlands eScience Center
+      ror: 00rbjv475
```
Comment (jspaaks) [inline schema-guide.md:4028]:
```suggestion
      ror: https://ror.org/00rbjv475
```

File: examples/1.3.0/pass/entity-author-with-ror/CITATION.cff [hunk 8]
```diff
@@ -0,0 +1,9 @@
+authors:
+- name: German Aerospace Center
+  ror: 04bwf3e34
```
Comment (jspaaks) [inline examples/1.3.0/pass/entity-author-with-ror/CITATION.cff:3]:
```suggestion
  ror: https://ror.org/04bwf3e34
```

File: schema.json [hunk 9]
```diff
@@ -1843,6 +1847,18 @@
             "$ref": "#/$defs/strictish-string",
             "description": "A region."
         },
+        "ror": {
+            "$comment": "Regex from 'https://github.com/ror-community/ror-schema/blob/a985c6e97293c0ebcc50aae89a2092c557c1dbec/ror_schema_v2_0.json#L116'",
+            "description": "An identifier for a research organization. See https://ror.org/ for more information.",
+            "examples": [
+                "https://ror.org/04bwf3e34",
+                "00rbjv475"
```
Comment (jspaaks) [inline schema.json:1855]:
```suggestion
                "https://ror.org/00rbjv475"
```

File: schema.json [hunk 10]
```diff
@@ -1843,6 +1847,18 @@
...
         },
+        "ror": {
+            "$comment": "Regex from 'https://github.com/ror-community/ror-schema/blob/a985c6e97293c0ebcc50aae89a2092c557c1dbec/ror_schema_v2_0.json#L116'",
+            "description": "An identifier for a research organization. See https://ror.org/ for more information.",
+            "examples": [
+                "https://ror.org/04bwf3e34",
+                "https://ror.org/00rbjv475"
+            ],
+            "maxLength": 9,
+            "minLength": 9,
```
Comment (jspaaks) [inline schema.json:1858]:
```suggestion
            "maxLength": 25,
            "minLength": 25,
```

File: schema-guide.md [hunk 11]
```diff
@@ -3998,6 +4014,20 @@ When there are multiple licenses, it is assumed their relationship is OR, not AN
         region: Renfrewshire
     ```
 
+### `definitions.ror`
+
+- **type**: string of length 9 with pattern `^0[0-9|a-z]{8}$`
```
Comment (jspaaks) [inline schema-guide.md:4019]:
```suggestion
- **type**: string of length 25 with pattern `^https://ror.org/0[0-9|a-z]{8}$`
```

File: schema.json [hunk 12]
```diff
@@ -351,6 +351,10 @@
                     "$ref": "#/$defs/region",
                     "description": "The entity's region."
                 },
+                "ror": {
+                    "$ref": "#/$defs/ror",
+                    "description": "The entity's Research Organization Registry (ror) identifier."
```
Comment (sdruskat) [inline schema.json:356]:
```suggestion
                    "description": "The entity's Research Organization Registry (ROR) identifier."
```

**Review by sdruskat — APPROVED** [review]
Good to go, one miniscule orthographical thing.

**Pull request merged**